strassengezwitscher/contact/forms.py

```diff
@@ -0,0 +1,31 @@
+from django import forms
+
+
+class ContactForm(forms.Form):
+    name = forms.CharField(required=False, max_length=50)
+    email = forms.EmailField(required=False)
+    subject = forms.CharField()
+    message = forms.CharField(widget=forms.Textarea)
+    journalist = forms.BooleanField(required=False)
+    confidential = forms.BooleanField(required=False)
+    files = forms.FileField(required=False, max_length=50, widget=forms.ClearableFileInput(attrs={'multiple': True}))
+
+    @classmethod
+    def files_are_valid(cls, request):
+        """Takes a request and returns True if the uploaded files are valid. Otherwise returns found errors."""
+        uploaded_files = request.FILES.getlist('files')
+        if len(uploaded_files) >= 2:
+            # Django's validation does only work for a single file. When uploading multiple files only the last one gets
+            # checked for validity. We could subclass FileField to suit our needs. Somebody did that already:
+            # https://github.com/Chive/django-multiupload
+            # However, the pain of the following workaround is not strong enough to add another dependency.
+            # As we have multiple files we create a new form with the same data but only a single file for each file
+            # and check the new form's validity.
+            # Another alternative would be not relying on Django's automatic FileField validation at all and just do
+            # that tiny bit of validation ourselves for all files in request.FILES.
+            for uploaded_file in uploaded_files:
+                request.FILES['files'] = uploaded_file
+                temp_form = cls(request.POST, request.FILES)
+                if not temp_form.is_valid():
+                    return temp_form.errors
```

Comment:
Mh, klingt nicht verkehrt, aber: Djangos Validation kommt mit mehreren Files nicht klar, darum obiger Workaround.
Wenn dieser Workaround aktiv wird, dann kann ich in die Liste der Errors des `files`-Feldes durchaus die Fehler mehrerer Dateien aggregieren. Der Haken dabei ist: Der Workaround wird nur aktiv, wenn Django nicht von sich aus schon gemeckert hat. Das passiert, wenn die erste hochgeladene Datei nicht valide ist, weil diese nämlich von Django geprüft wird.
Wenn ich eine Zusammenfassung einbauen würde, dann hängt die Anzahl der zusammengefassten Fehler ab von der Reihenfolge der Auswahl der Anhänge. Das könnte sich sehr inkonsistent anführen.
Stattdessen würde ich es dann lieber lassen, wie es ist: Der Nutzer bekommt die ValidationErrors scheibchenweise, wenn er es tatsächlich schafft, die Validation an dieser Stelle zu triggern.